src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -16,18 +34,23 @@ export const TodoFilter = () => (
         type="text"
         className="input"
         placeholder="Search..."
+        value={valueInput}
+        onChange={(ev) => onChangeInput(ev.target.value)}
       />
       <span className="icon is-left">
         <i className="fas fa-magnifying-glass" />
       </span>
 
       <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
-        />
+        {valueInput && (
```

Comment:
I'd move `span` wrapper inside this condition as well (at least it'll fix one test)